### PR TITLE
use new next features, remove env var (to allow builds)

### PR DESCRIPTION
**Pull request by vassbence**
cc @timothyis

### CODE REVIEW

File: pages/themes/newest.js [hunk 1]
```diff
@@ -1,14 +1,23 @@
...
+    <PluginThemeShowcase plugins={themes} variant="theme" filter="newest" />
+  )
+}
+
+export function getStaticProps() {
+  // get all featured plugins then get the preview image's relative src
+  const themes = allPlugins
     .filter((p) => p.type === 'theme')
     .sort((a, b) =>
       a.dateAdded < b.dateAdded ? 1 : a.dateAdded > b.dateAdded ? -1 : 0
```
Comment (pacocoursey) [inline pages/themes/newest.js:15]: Does this need the preview image?
```
      preview: getPluginPreviewImage(p.name),
```

File: yarn.lock [hunk 2]
```diff
@@ -2,62 +2,77 @@
 # yarn lockfile v1
 
 
-"@ampproject/toolbox-core@^2.3.0":
-  version "2.3.0"
-  resolved "https://registry.yarnpkg.com/@ampproject/toolbox-core/-/toolbox-core-2.3.0.tgz#f27bd17e01fdc6725c440aefa844f63466c0f37e"
-  integrity sha512-NT+kVR5Rm2cxp12h40IXgPRWmq0cpUdmcgZmgdelplp/q//4aWkt2+llGHR2foQJkwICxMVVlb/XidsHz0Rh9g==
+"@ampproject/toolbox-core@^2.6.0":
```
Comment (pacocoursey) [inline yarn.lock:5]: Why do we have both `package-lock` and `yarn-lock`?

File: pages/themes/newest.js (hunk 1 above)
Comment (vassbence) [reply pages/themes/newest.js:15]: The comment there is misleading but the code is correct, this page is the "list" view for the newest themes, there are no preview images present here. Removed the comment.

File: components/plugin/index.js [hunk 3]
```diff
@@ -1,14 +1,13 @@
...
 const MAX_COLORS = 4
 
 export default ({ featured, name, description, colors, preview, query }) =>
   featured ? (
-    <div
-      className={styles.featured}
-      style={{ backgroundImage: `url(${preview})` }}
-    >
+    <div className={styles.featured}>
+      <Image className={styles.featuredImage} src={preview.src} layout="fill" />
```
Comment (pacocoursey) [inline components/plugin/index.js:10]: We need to request the image in a specific, smaller, size. Right now the plugins page is loading 2.6mb+ images, which is not reasonable.

Comment (vassbence) [reply components/plugin/index.js:10]: Is there a way to do this through next/image?

Comment (vassbence) [reply components/plugin/index.js:10]: I tried forcing next to always load the 640px wide (smallest default device size) image by adding sizes="1px" (used 1px bc using the actual size of the image on the screen ~400px caused the browsers to pick a larger image due to retina screen), but the actual size of the gifs did not get smaller, maybe the _next/image endpoint is not that good at making gifs smaller?

Comment (vassbence) [reply components/plugin/index.js:10]: I could take a screenshot of those gifs and use that as preview images, or should i just make the resolution of the raw gif files smaller?

Comment (pacocoursey) [reply components/plugin/index.js:10]: Can we not run the gifs through `next/image` and just continue serving them via GitHub URLs?

**Review by pacocoursey — COMMENTED** [review]
Gifs are still taking around 10s to process from a cold hit. Looks like it's converting it to an animated webp?

Screen Shot 2020-12-09 at 12 41 49 PM

File: components/plugin/index.js [hunk 4]
```diff
@@ -1,14 +1,19 @@
...
-      className={styles.featured}
-      style={{ backgroundImage: `url(${preview})` }}
-    >
+    <div className={styles.featured}>
+      <Image
+        src={preview.src}
+        layout="fill"
+        objectFit="cover"
+        objectPosition="center top"
+        unoptimized
```
Comment (pacocoursey) [inline components/plugin/index.js:15]: Seems the gifs on plugin pages aren't unoptimized, i.e.: https://hyper-site-git-main.vercel.vercel.app/store/hyper-search

**Comment by vassbence** [comment]
All GIFs are now rendered without using next/image

**Comment by vercel-team** [comment]
#### 📝Changed routes:


| [**/**](https://hyper-site-72qhrty02.vercel.app/) | [**/plugins**](https://hyper-site-72qhrty02.vercel.app/plugins) |
|:-:|:-:|
|<a href="https://hyper-site-72qhrty02.vercel.app/">Screenshot of /</a><br /><sup><a href="https://images.weserv.nl?url=https://s.microlink.io/crpgMzlTZPfDgDecLxrwR25AdII_.png">(view full size)</a> |<a href="https://hyper-site-72qhrty02.vercel.app/plugins">Screenshot of /plugins</a><br /><sup><a href="https://images.weserv.nl?url=https://s.microlink.io/qBO6KcJTjUf9JQZkWFhauz0Ajmk_.png">(view full size)</a> |



| [**/plugins/newest**](https://hyper-site-72qhrty02.vercel.app/plugins/newest) | [**/store/\[name\]**](https://hyper-site-72qhrty02.vercel.app/store/[name]) |
|:-:|:-:|
|<a href="https://hyper-site-72qhrty02.vercel.app/plugins/newest">Screenshot of /plugins/newest</a><br /><sup><a href="https://images.weserv.nl?url=https://s.microlink.io/Y6qdUMI_rb2c8XndIZN1srcBVDw_.png">(view full size)</a> |<a href="https://hyper-site-72qhrty02.vercel.app/store/[name]">Screenshot of /store/[name]</a><br /><sup><a href="https://images.weserv.nl?url=https://s.microlink.io/MzWWyw_l2jm1qGcZew1wGLeN6oA_.png">(view full size)</a> |



| [**/store/\[name\]/source**](https://hyper-site-72qhrty02.vercel.app/store/[name]/source) | [**/themes**](https://hyper-site-72qhrty02.vercel.app/themes) |
|:-:|:-:|
|<a href="https://hyper-site-72qhrty02.vercel.app/store/[name]/source">Screenshot of /store/[name]/source</a><br /><sup><a href="https://images.weserv.nl?url=https://s.microlink.io/yKK_M3ComVDDmv_kUjPKJR4btYU_.png">(view full size)</a> |<a href="https://hyper-site-72qhrty02.vercel.app/themes">Screenshot of /themes</a><br /><sup><a href="https://images.weserv.nl?url=https://s.microlink.io/CNJEkJzuzq64Zpb2B1gn1is3p70_.png">(view full size)</a> |



And 1 other route:
- [**/themes/newest**](https://hyper-site-72qhrty02.vercel.app/themes/newest)



Commit ce4193bacafda0a6436d3f0ec9659386e0aee0f9 (https://hyper-site-72qhrty02.vercel.app).

**Review by pacocoursey — APPROVED** [review]

**Pull request merged**